src/openrct2/world/Map.cpp

```diff
@@ -1539,6 +1540,54 @@ void map_update_tiles()
     }
 }
 
+void map_calc_highest_tile_height()
+{
+    gMapHighestTileHeight = 0;
+    for (int32_t i = 0; i < MAX_TILE_TILE_ELEMENT_POINTERS; i++)
+    {
+        TileElement* tile = &gTileElements[i];
```

Comment:
It is better to have a double for loop for y axis and x axis and then use `map_get_surface_element`. The surface element is not always going to be the first element on a tile and some tiles do not have any elements at all so you need to check for `nullptr`.